web-animations-1/Overview.bs

```diff
@@ -833,14 +833,58 @@ follows:
     <var>animation</var>, if any.
 1.  If <var>new timeline</var> is the same object as <var>old timeline</var>,
     abort this procedure.
+1.  Let |previous play state| be |animation|'s [=play state=].
+1.  Let |previous current time| be the |animation|'s [=current time=].
+1.  Let |had finite timeline| be true if |animation| has an associated
+    <a>timeline</a> that is not [=monotonically increasing=].
 1.  Let the <a>timeline</a> of <var>animation</var> be <var>new timeline</var>.
+1.  Perform the steps corresponding to the <em>first</em> matching
+    condition from the following, if any:
+
+    <div class="switch">
+
+    :   If |animation| has an associated
+        <a>timeline</a> that is not [=monotonically increasing=],
+
+    ::  1.  [=Apply any pending playback rate=] on |animation|
+        1.  Let |seek time| be zero if [=playback=] rate &ge; 0, and
+            |animation|'s <a>associated effect end</a> otherwise.
+        1.  Update the [=start time=] and [=hold time=] based on the first
+            matching condition if any
+
+            <div class="switch">
+
+            :   If <em>either</em> of the following conditions are true:
+                *    |previous playback rate| is 'running' or,
+                *    |previous playback rate| is 'finished'
+
+            ::  Set |animation|'s [=start time=] to |seek time|.
+
+            :   If <em>both</em> of the following conditions are true:
```

Comment:
Here we are changing current time of paused animation if the new timeline is active and leaving unchanged otherwise. 
I think it's unexpected for the paused animation to change current time. 
I suggest leaving the old hold_time unchanged when setting new timeline to paused animation.

Reply:
If we don't preserve current time, we cause a jump in the progress of the animation when transition from a scroll timeline to a document timeline.  Here we are setting current time to preserve the current progress in the animation.

Reply:
Just to followup on my last comment.  We are preserving current time if we can, and leaving it along otherwise.  Transitioning between scroll timelines is handled in the previous branch.  AFAIK, the only way for the document timeline to be inactive is if the animation clock has not ticked yet.  If the new timeline is null, it will simply preserve the previous scroll timeline's value of hold time.

Handling in this manner, no distinction needs to be made regarding whether the animation was running at the time of the timeline change.